MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java

```diff
@@ -0,0 +1,184 @@
+import static com.mapbox.mapboxsdk.style.layers.Property.TEXT_ANCHOR_TOP;
+import static com.mapbox.mapboxsdk.style.layers.Property.TEXT_JUSTIFY_AUTO;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textField;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textJustify;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textRadialOffset;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textSize;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textVariableAnchor;
+
+
+/**
+ * To increase the chance of high-priority labels staying visible, provide the map
+ * renderer a list of preferred text anchor positions via
+ * {@link com.mapbox.mapboxsdk.style.layers.PropertyFactory#textVariableAnchor(String[])}.
+ */
+public class VariableLabelPlacementActivity extends AppCompatActivity {
+
+  private static final String GEOJSON_SRC_ID = "poi_source_id";
+  private static final String POI_LABELS_LAYER_ID = "poi_labels_layer_id";
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_style_variable_text_placement);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+        mapboxMap.setStyle(new Style.Builder().fromUrl(Style.LIGHT)
+                .withSource(new GeoJsonSource(GEOJSON_SRC_ID))
+                // Adds a SymbolLayer to display POI labels
+                .withLayer(new SymbolLayer(POI_LABELS_LAYER_ID, GEOJSON_SRC_ID)
+                    .withProperties(
+                        //iconImage(POI_ICON),
+                        textField(get("description")),
+                        textSize(17f),
+                        textColor(Color.RED),
+                        textVariableAnchor(
+                            new String[]{TEXT_ANCHOR_TOP, TEXT_ANCHOR_BOTTOM, TEXT_ANCHOR_LEFT, TEXT_ANCHOR_RIGHT}),
+                        textJustify(TEXT_JUSTIFY_AUTO),
+                        textRadialOffset(0.5f))),
+            new Style.OnStyleLoaded() {
+              @Override
+              public void onStyleLoaded(@NonNull final Style style) {
+                VariableLabelPlacementActivity.this.mapboxMap = mapboxMap;
+                new LoadGeoJson(style,VariableLabelPlacementActivity.this).execute();
```

Comment:
We are currently trying to load a local geojson file and aren't manipulating its contet. We can remove a lot of this boilerplate code if we rely on the loading map resources through assets. Us  `asset://poi_places.geojson` as url for the second parameter of the geojsonsource constructor. 

The json loader can be removed and the whole asynctask setup

Reply:
Got rid of the `AsyncTask` and `onDataLoaded()`.

Doing this didn't work for me (lol):
<img width="541" alt="Screen Shot 2019-05-22 at 4 51 53 PM" src="https://user-images.githubusercontent.com/4394910/58216151-ff8d0b80-7cb1-11e9-95e6-06ac679e08ee.png">

So, I used the `loadJsonFromAsset()` method that's used in some other examples:

https://github.com/mapbox/mapbox-android-demo/blob/fff1abe14114b4d7691996d1cba5be548c305e5d/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java#L71
https://github.com/mapbox/mapbox-android-demo/blob/fff1abe14114b4d7691996d1cba5be548c305e5d/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/VariableLabelPlacementActivity.java#L95-L109

Example's working the same as seen in the GIF in OP.

Did I misinterpret your instructions? Is there some even smoother way to pull from the assets folder?

Reply:
My bad, didn't double check the constructors. The issue with this setup is that you are trying to load that URI as an geojson string. 
I tried using the URL constructor but that failed as that URI is not a valid URL. 
Was able to make it work as:

```java
    source = new GeoJsonSource(MARKER_SOURCE);
    source.setUrl("asset://something.geojson");
```